File: android/app/src/main/java/com/mirego/csmapapplication/helper/Item.java
package com.mirego.csmapapplication.helper;

/**
 * Created by PainDeViande on 24/03/2018.
 */

public class Item {
    private String name, component, notes, type, address;
    private float lat, lon;

    public Item() {
    }

    public String getName() {
        return name;
    }

    public String getComponent() {
        return component;
    }

    public String getNotes() {
        return notes;
    }

    public String getType() {
        return type;
    }

    public String getAddress() {
        return address;
    }

    public float getLat() {
        return lat;
    }

    public float getLon() {
        return lon;
    }

    public Item(String name, String component, String notes, String type, String address, float lat, float lon) {
        this.name = name;
        this.component = component;
        this.notes = notes;
        this.type = type;
        this.address = address;
        this.lat = lat;
        this.lon = lon;

    }

}

File: android/app/src/main/java/com/mirego/csmapapplication/model/Item.kt
package com.mirego.csmapapplication.model

import com.google.gson.annotations.SerializedName

class Item {
    @SerializedName("name")
    private val name: String? = null
    @SerializedName("component")
    private val component: String? = null
    @SerializedName("notes")
    private val notes: String? = null
    @SerializedName("type")
    private val type: String? = null
    @SerializedName("lat")
    private val lat: Double? = null
    @SerializedName("lon")
    private val lon: Double? = null
    @SerializedName("address")
    private val address: String? = null
    @SerializedName("distance")
    private val distance: String? = null

    fun getType() : String = type?: ""
    fun getName() : String = name?: ""
    fun getDescription() : String = component?: ""
    fun getLon(): Double = Math.round((lon?:0.0) * 10000.0) / 10000.0
    fun getLat(): Double = Math.round((lat?:0.0) * 10000.0) / 10000.0
    fun getDistance(): String = (Math.round((distance?:"0.0").toDouble() * 100.0) / 100.0).toString()
    fun getLocation(): String = "Lat: " + (getLat()).toString() + "  Lon: " + (getLon()).toString()

}